Mathlib/Algebra/Quaternion.lean

```diff
@@ -148,7 +148,6 @@ theorem coe_injective : Function.Injective (coe : R → ℍ[R,c₁,c₂]) := fun
 theorem coe_inj {x y : R} : (x : ℍ[R,c₁,c₂]) = y ↔ x = y :=
   coe_injective.eq_iff
 
--- Porting note: removed `simps`, added simp lemmas manually
 instance : Zero ℍ[R,c₁,c₂] := ⟨⟨0, 0, 0, 0⟩⟩
 
 @[simp] theorem zero_re : (0 : ℍ[R,c₁,c₂]).re = 0 := rfl
```

Comment:
I think we shouldn't be removing the porting note here. We should fix `simps` instead!

Reply:
It's not that `simps` doesn't generate the lemmas, it's that they have bad names because this is an `instance` (at least, that was my read of the situation after confirming that `simps` can generate them).

Reply:
Then that's still `simps`' fault, right? It should know how to correctly name the lemmas coming from the projections of an instance. It did in Lean 3

Reply:
I thought maybe we could fix this by giving a name to the instance:
```lean
@[simps?]
instance zero : Zero ℍ[R,c₁,c₂] := ⟨⟨0, 0, 0, 0⟩⟩
```
But now `@[simps]` calls the lemmas `zero_zero_re`, etc. :(

Reply:
Yeah, if it could be fixed by naming the instance, that would be great, but we *definitely* shouldn't be having `simps` name something based on the automatic naming of instances.

Reply:
Yes, `simps` should just ignore the instance name part

Reply:
I've put these notes back, but added a statement that we should adjust `simps` to name properly.